integration/base/src/main/java/com/inrupt/client/integration/base/AccessGrantScenarios.java

```diff
@@ -581,6 +613,69 @@ void accessGrantCreateNonRdfTest(final Session session) throws IOException {
         assertDoesNotThrow(accessGrantClient.delete(grant).toCompletableFuture()::join);
     }
 
+    private static void prepareACPofResource(final SolidSyncClient authClient, final URI resourceURI) {
+
+        final IRI acpAllOf = rdf.createIRI(ACP.allOf.toString());
+        final IRI acpVc = rdf.createIRI(ACP.vc.toString());
+        final IRI acpAllow = rdf.createIRI(ACP.allow.toString());
+        final IRI acpApply = rdf.createIRI(ACP.apply.toString());
+        final IRI acpAccessControl = rdf.createIRI(ACP.accessControl.toString());
+        final IRI aclRead = rdf.createIRI(ACL.Read.toString());
+        final IRI aclWrite = rdf.createIRI(ACL.Write.toString());
+
+        // find the acl Link in the header of the resource
+        final Request req = Request.newBuilder(resourceURI)
+                .HEAD()
+                .build();
+        final Response<Void> res = authClient.send(req, Response.BodyHandlers.discarding());
+        final Headers.Link acrLink = res.headers().allValues("Link").stream()
+            .flatMap(l -> Headers.Link.parse(l).stream())
+            .filter(link -> link.getParameter("rel").contains("acl"))
+            .findAny()
+            .orElse(null);
+
+        // add the triples needed for access grant
+        if (acrLink != null) {
+            final URI resourceACRurl = acrLink.getUri();
+            final IRI resourceACRiri = rdf.createIRI(resourceACRurl.toString());
+
+            final Dataset dataset = rdf.createDataset();
+            //read the existing triples and add them to the dataset
+            try (final SolidRDFSource resource = authClient.read(resourceACRurl, SolidRDFSource.class)) {
```

Comment:
Note that `SolidRDFSource` implements `Dataset`, so you could omit the `rdf.createDataset()` above and just call `resource.add(...)`. Then, you can just call `authClient.update(resource);`